code/modules/mob/living/simple_animal/parrot.dm

```diff
@@ -525,12 +528,12 @@
 			return
 
 		if(Adjacent(parrot_perch))
-			src.loc = parrot_perch.loc
-			drop_held_item()
-			parrot_state = PARROT_PERCH
-			icon_state = "parrot_sit"
-			return
-
+			if(name = "Clockwork Hawk")

```

Comment:
Why are you checking name here? This will prevent any parrot not named this from perching.